Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1

```diff
@@ -737,6 +767,22 @@ function Start-Stop-Recycle-ApplicationPool {
     $appCmdArgs = [string]::Format('{0} apppool /apppool.name:"{1}"', $action, $appPoolName)
     $command = "`"$appCmdPath`" $appCmdArgs"
 
+    $appPoolState = Get-AppPoolState -appPoolName $appPoolName
+
+    if ($appPoolState)
+    {
+        $isAppPoolInStartedState =  ($appPoolState -like "Started")
+
+        if (($isAppPoolInStartedState -and $action -like "start") -or ((-not $isAppPoolInStartedState) -and $action -like "stop")) {
```

Comment:
Is the assumption that not started means "stop" state. Are there other states like "Paused" or "Starting" etc?

Reply:
There are other possible states. changed logic.